music21/stream/tools.py

```diff
@@ -131,12 +134,26 @@ def removeDuplicates(thisStream: stream.Stream,
         {0.0} <music21.note.Note C>
         {1.0} <music21.bar.Barline type=final>
 
+    As the example shows, this function defaults to working on a stream inPlace.
+
+    >>> testInPlace == s
+    True
+
+    To avoid this, set inPlace to False.
+
+    >>> testNotInPlace = stream.tools.removeDuplicates(s, inPlace=False)
+    >>> testNotInPlace == s
+    False
+
     '''
 
     supportedClasses = [meter.TimeSignature, key.KeySignature, clef.Clef]
 
     listOfObjectsToRemove = []
 
+    if not inPlace:
+        thisStream = deepcopy(thisStream)
```

Comment:
This way you get the derivation set also.
```suggestion
        thisStream = thisStream.coreCopyAsDerivation('removeDuplicates')
```